File: abc/131/c.py
from fractions import gcd

# 入力
A, B, C, D = map(int, input().split())


# CまたはDで割り切れる数を求め、それを差し引いて求める
def f(x):
    return x - (x // C + x // D - x // (C * D // gcd(C, D)))


ans = f(B) - f(A - 1)

# 出力
print(ans)

File: abc/131/c.rs
#![allow(non_snake_case)]
#![allow(unused_variables)]
#![allow(dead_code)]

fn main() {
    let (A, B, C, D): (i64, i64, i64, i64) = {
        let mut line: String = String::new();
        std::io::stdin().read_line(&mut line).unwrap();
        let mut iter = line.split_whitespace();
        (
            iter.next().unwrap().parse().unwrap(),
            iter.next().unwrap().parse().unwrap(),
            iter.next().unwrap().parse().unwrap(),
            iter.next().unwrap().parse().unwrap(),
        )
    };

    fn gcd(a: i64, b: i64) -> i64 {
        if b == 0 {
            a
        } else {
            gcd(b, a % b)
        }
    }

    let f = |x| x - (x / C + x / D - x / (C / gcd(C, D) * D));
    let ans = f(B) - f(A - 1);

    println!("{}", ans);
}
